src/k8s/pkg/k8sd/api/cluster_join.go

```diff
@@ -19,15 +20,20 @@ func postClusterJoin(m *microcluster.MicroCluster, s *state.State, r *http.Reque
 		return response.BadRequest(fmt.Errorf("failed to parse request: %w", err))
 	}
 
+	hostname, err := utils.CleanHostname(req.Name)
+	if err != nil {
+		return response.BadRequest(fmt.Errorf("invalid hostname %q: %w", req.Name, err))
+	}
+
 	// differentiate between control plane and worker node tokens
 	info := &types.InternalWorkerNodeToken{}
 	if info.Decode(req.Token) == nil {
 		// valid worker node token
-		if err := m.NewCluster(req.Name, req.Address, map[string]string{"workerToken": req.Token}, time.Second*180); err != nil {
+		if err := m.NewCluster(hostname, req.Address, map[string]string{"workerToken": req.Token}, time.Second*180); err != nil {
```

Comment:
The timeout should depend on the deadline of the request (if any) and not be a fixed to 180s. Similar to [here](https://github.com/canonical/k8s-snap/blob/main/src/k8s/pkg/k8s/client/cluster.go#L34-L39)

(I know this is not directly part of this PR but noticed it since this line was updated)